### ISSUE TITLE
Constantly getting error in console `WebSocket connection to 'ws://localhost:8081/' failed:`

### ISSUE THREAD

**Issue by IzaGz** [Jonghakseo/chrome-extension-boilerplate-react-vite#477]
Open console and see this error:
`Reload server disconnected.
Please check if the WebSocket server is running properly on ws://localhost:8081. This feature detects changes in the code and helps the browser to reload the extension or refresh the current tab.`

<img width="520" alt="Снимок экрана 2024-05-22 в 01 43 07" src="https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/assets/19924220/3f257883-c282-4eba-9803-f2b3d3a9ef2e">

also in messages (not errors):
<img width="394" alt="Снимок экрана 2024-05-22 в 01 48 43" src="https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/assets/19924220/ee7e8ad2-fa0e-4071-84a4-0127f0321d5f">

This clogs the console, not convenient.
How to fix this?

**Comment by Jonghakseo** [comment]
Plz check your dev-server

```bash
$ pnpm dev-server
```

This is automatic on `pnpm dev` in latest version

- https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/pull/467

**Comment by IzaGz** [comment]
i am trying latest version , starting it as `pnpm dev` but still getting this error, 
i have tried to run the `pnpm dev` in separate terminal and getting error that this is already running

**Comment by ulises-castro** [comment]
Same issue

**Comment by IzaGz** [comment]
still getting this 
@PatrykKuniczak @Jonghakseo  Can you please, reaopen this issue?

<img width="1054" alt="Снимок экрана 2024-06-11 в 00 54 16" src="https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/assets/19924220/9ed1b2eb-5e50-419b-8114-e4d575674e68">

**Comment by Jonghakseo** [comment]
Have you checked to see if there are any other processes already occupying port 8081, and would like some information about the operating environment (OS, node version, etc.)?

**Comment by IzaGz** [comment]
@Jonghakseo there is no other processes running the 8081, only dev-server starting ont this port, macOS Sonoma 14.5 , 
node -v 20.13.1

**Comment by Jonghakseo** [comment]
Have you tried reloading the extension?

**Comment by IzaGz** [comment]
@Jonghakseo 
Yes! Of course

**Comment by ulises-castro** [comment]
> @Jonghakseo there is no other processes running the 8081, only dev-server starting ont this port, macOS Sonoma 14.5 , node -v 20.13.1

I'm on Sonoma 14.5, and node v21.
I had tried reloading the chrome extension but it didn't fix it, maybe there is a new firewall policy about websocket or maybe something related to Nodejs?

**Comment by Jonghakseo** [comment]
I've tried the same node version on sonoma 14.5, but it doesn't reproduce the issue, which is interesting...

**Comment by Jonghakseo** [comment]
Can I know the browser information? I'm curious about your Chrome version.

**Comment by PatrykKuniczak** [comment]
@Jonghakseo I have the same problem on Windows, version of Chrome is 126.0.6478.127
![image](https://github.com/user-attachments/assets/b11ebb09-b1b1-454c-8780-54ebce82cd8f)

**Comment by Jonghakseo** [comment]
Exactly the same version, but not reproducible... It's not like you're running more than one boilerplate project, right?

<img width="240" alt="스크린샷 2024-07-14 오후 10 13 53" src="https://github.com/user-attachments/assets/dcf7fbba-27ca-4dfb-b6fa-6352b78bf2c7">

**Comment by PatrykKuniczak** [comment]
@Jonghakseo No, one 

But there's some strange things when i stop dev script and try to rerun, sometimes i have an info about "On this port sth running already" so kind of this.

No i struggling with this, and i was tried to restart PC and i hope it work now.
I tried to kill task on this port but after few seconds that task become active again ... xD

**Comment by PatrykKuniczak** [comment]
What's going on here
![image](https://github.com/user-attachments/assets/4e71b832-be30-47cb-9790-0666d619055d)

I can't shut down this process.

OOO i'm lucky i do some chances in code, and i can run dev script, it's crazzy.

**Comment by PatrykKuniczak** [comment]
@Jonghakseo OO probably i see what's going on.

If i get any error, then script break and when i trying to rerun manually after error, that error about running process on port occur.

Maybe it's related

**Comment by Jonghakseo** [comment]
I'll have to take a look at the code and issues in other HMR projects that use websockets... 
I don't have a clue yet.

**Comment by thavguard** [comment]
same issue 
![image](https://github.com/user-attachments/assets/60260bbf-ad63-4a24-8bec-0984d4a803ca)

**Comment by MassDo** [comment]
Same issue

**Comment by Jonghakseo** [comment]
To those who reported this issue...

Is HMR not working properly due to this issue, or is it working properly after reloading the extension?

If you were developing with `pnpm dev` and shut down the local running dev server, it's normal to see the above message, and it should stop when the dev server starts up again.

I'd like to know more specifically if the above issue is interfering with your development.

**Comment by IzaGz** [comment]
> Is HMR not working properly due to this issue, or is it working properly after reloading the extension?'

now i have downloaded latest project version, tried to start it and getting same issue, i have tried to reload and rebuild it, but it is still present

> If you were developing with pnpm dev and shut down the local running dev server, it's normal to see the above message, and it should stop when the dev server starts up again.

no, i am getting thouse errors when server is running,
i am using latest chrome version

**Comment by Jonghakseo** [comment]
We're looking for help from people who can reproduce this issue on the Discord channel. 

https://discord.com/channels/1263404974830915637/1264515166473551906/1270506301763620915

**Comment by PatrykKuniczak** [comment]
Max we can do for now is:
```ts
  ws.onerror = () => {
    console.clear();
  };
```

i was trying to solve this entire yesterday, but it's not able to catch or sth like that.

There's issue about that:
https://github.com/websockets/ws/issues/1537

I was waiting for response, that's interesting case.

**Comment by jjwallace** [comment]
![image](https://github.com/user-attachments/assets/f20789e7-59bd-4b1c-8d5b-d51af22bbde7)

I also get the same in Ubuntu with Brave and now with the pnpm dev command I get the above ws problem

**Comment by PatrykKuniczak** [comment]
> ![image](https://private-user-images.githubusercontent.com/4450959/358924717-f20789e7-59bd-4b1c-8d5b-d51af22bbde7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwMDIwODYsIm5iZiI6MTcyNDAwMTc4NiwicGF0aCI6Ii80NDUwOTU5LzM1ODkyNDcxNy1mMjA3ODllNy01OWJkLTRiMWMtOGQ1Yi1kNTFhZjIyYmJkZTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MThUMTcyMzA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Nzg5MTdjNDZmYzliNTE3ZDhmNTE3MTRhNWIxNDRmY2VjMTA5YWEzMmZkYjRlZDEwOTU4OWY3NjNlOTE5MTk4YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.J0Z4AFvHCPqaGPKXly9H7-rnM4B6uLXiiWNpcyLNh3w)
> 
> I also get the same in Ubuntu with Brave and now with the pnpm dev command I get the above ws problem. Port is open, restarted machine and everything.
> 
> Update: deleting lock files fixed my issue, still get the ws console log spamming

Yeah, that's issue between package versions, cause i recently change 'ws' from deps to 'devDeps'

That's why remove node_module script is necessary @Jonghakseo :)

We need to implement it, for upgrade `pnpm clean` 😄

**Comment by PatrykKuniczak** [comment]
@jjwallace We cannot do nothing with it, i spent 2 days or trying to find solution, but nothing works.

I'm waiting for https://github.com/Jonghakseo/chrome-extension-boilerplate-react-vite/issues/477#issuecomment-2290868107

**Comment by ulises-castro** [comment]
I was using Brave (latest version), then I switch to Microsoft Edge and this issue gone. 
I'm not sure why it does not work ws properly in Brave yet.

**Comment by PatrykKuniczak** [comment]
#719 Doesn't solve this issue, it still happening, i think it's issue browser related.

I was reading a little about it and it's probably because of security issue.

I think, we can close it here.

But for people which will read this thread in the future.
Feel free to read here, if this issue will be very annoying for you, we will try one more time to create solution for this, maybe some kind of custom thing.

**Comment by PatrykKuniczak** [comment]
It can't be fixed in secure way, i need to close it.
